modular_skyrat/modules/new_meteors/code/new_meteors.dm

```diff
@@ -0,0 +1,53 @@
+#define METEOR_WAVE_MIN_NOTICE 210
+#define METEOR_WAVE_MAX_NOTICE 240
+#define METEOR_WAVE_DURATION 37
+
+/datum/round_event_control/meteor_wave
+	///Where will the meteors be coming from? -- Established in admin_setup, passed down to round_event
+	var/start_side
+
+/datum/round_event/meteor_wave
+	///Which direction the storm will come from.
+	var/start_side
+
+/datum/round_event/meteor_wave/New()
+	..()
+	start_when = rand(METEOR_WAVE_MIN_NOTICE, METEOR_WAVE_MAX_NOTICE)
+	end_when = start_when + METEOR_WAVE_DURATION
+
+/datum/round_event_control/meteor_wave/admin_setup()
+	if(!check_rights(R_FUN))
+		return ADMIN_CANCEL_EVENT
+
+	var/force_dir = tgui_alert(usr, "Choose a side to throw rocks at?", "Interrupt some ERP?", list("Yes", "No", "Cancel"))
+	if(force_dir == "Cancel")
+		return ADMIN_CANCEL_EVENT
+
+	if(force_dir == "Yes")
+		var/chosen_direction = tgui_input_list(usr, "Pick one!","Plausible Deniability.", list("North", "South", "East", "West"))
+		switch(chosen_direction)
+			if("North")
+				start_side = NORTH
+			if("South")
+				start_side = SOUTH
+			if("East")
+				start_side = EAST
+			if("West")
+				start_side = WEST
+
+/datum/round_event/meteor_wave/announce(fake)
+	var/datum/round_event_control/meteor_wave/meteor_event = control
+	if(meteor_event.start_side)
+		start_side = meteor_event.start_side
+	else
+		start_side = pick(GLOB.cardinals)
+
+	priority_announce("Meteors have been detected on collision course with the station. The early warning system estimates first collision in approximately [start_when * 2] seconds, coming from the [dir2text(start_side)]. Ensure all sensitive areas and equipment are shielded.", "Meteor Alert", ANNOUNCER_METEORS)
```

Comment:
For the sake of consistency with all other events like this, having the side be a ship direction rather than a cardinal direction would be best (Fore, Aft, Port, Starboard)

Reply:
Sandstorm uses up/down/left/right?

Reply:
Do we even have a proc that takes the integer and turns it into cardinal directions?

Reply:
Sandstorm's announcement reads as follows:
```
/datum/round_event/sandstorm/announce(fake)
	var/datum/round_event_control/sandstorm/sandstorm_event = control
	if(sandstorm_event.start_side)
		start_side = sandstorm_event.start_side
	else
		start_side = pick(GLOB.cardinals)

	var/start_side_text = "unknown"
	switch(start_side)
		if(NORTH)
			start_side_text = "fore"
		if(SOUTH)
			start_side_text = "aft"
		if(EAST)
			start_side_text = "starboard"
		if(WEST)
			start_side_text = "port"
		else
			stack_trace("Sandstorm event given [start_side] as unrecognized direction. Cancelling event...")
			kill()
			return

	priority_announce("A large wave of space dust is approaching from the [start_side_text] side of the station. \
	etc etc etc
```
it converts start side into start_side_text in relation to the nautical direction, rather than cardinal

I wish it was a proc. TG could probably use a proc for it...

Reply:
I'll throw this in soon.